### PR TITLE
Adapt `ps_specials` for multiples versions of PrestaShop

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Adapt `ps_specials` for multiples versions of PrestaShop
| Type?         | refacto
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Relative to PrestaShop/Prestashop#28988
| How to test?  | Try this module in version `< 1.7.4.0` and in recent version (`8.0.0`)

### CODE REVIEW

**Review by kpodemski — CHANGES_REQUESTED** [review]
Maybe we could simply change the minimum required PS version? There are no other significant updates for this module

**Comment by Progi1984** [comment]
@kpodemski I have done changes for being compatible with old versions. We can change the minimum required PS version when it will be really useful.

**Review by kpodemski — APPROVED** [review]

**Review by khouloudbelguith — COMMENTED** [review]
Hello @Progi1984,

I tried this PR with those versions

- 178x + PHP 7.2 => OK
- 8.0.x + php8.1 => ok
- PS1712 + php7.1 => NOK, after installing the PR, I have an exception in the FO
image
- PS1725 + PHP 7.1 => NOK after installing the PR, I have an exception in the FO
image
- PS1744 + php7.0 => NOK, after installing the PR, I have an exception in the module manager configure page & in the FO 
I attached a screen record

https://user-images.githubusercontent.com/16067358/184408455-fd5fe8c7-5641-4251-8de0-16b022dfeb8a.mp4

Thanks!

**Review by zuk3975 — APPROVED** [review]

**Review by leemyongpakvn — COMMENTED** [review]
I guess L229 must be
if (version_compare(`_PS_VERSION_`, '1.7.4', '>=')) {

**Review by matks — APPROVED** [review]

**Comment by leemyongpakvn** [comment]
@matks I think there is a mistake at [Line 229](https://github.com/PrestaShop/ps_specials/pull/10/files/b0b5934bd9c351d0a13db03a5556ce68f9922202#diff-d5f68d478680ed7436708049fda08f577aef7ca9b5c5cd9284debb4b03125035R229) because we have no **PHP_VERSION** 1.7.4

File: ps_specials.php [hunk 1]
```diff
@@ -227,15 +226,27 @@ private function getSpecialProducts()
...
-        $presenter = new ProductListingPresenter(
-            new ImageRetriever(
-                $this->context->link
-            ),
-            $this->context->link,
-            new PriceFormatter(),
-            new ProductColorsRetriever(),
-            $this->context->getTranslator()
-        );
+        if (version_compare(PHP_VERSION, '1.7.4', '>=')) {
```
Comment (kpodemski) [inline ps_specials.php:229]:
```suggestion
        if (version_compare(_PS_VERSION_, '1.7.4', '>=')) {
```

**Review by kpodemski — DISMISSED** [review]

**Comment by kpodemski** [comment]
@leemyongpakvn you're right! thanks

**Review by Hlavtox — APPROVED** [review]

**Review by kpodemski — APPROVED** [review]

**Review by florine2623 — COMMENTED** [review]
Hello @Progi1984 ,

Tested with the following PS versions and PHP 7.1 and 7.4 :
- PS 1.7.3.2 ✅ 
- PS 1.7.5.2 ✅ 
- PS 1.7.8.8 ✅ 
- PS 8.0.1 ✅ 
- PS develop ✅ 
- PS 1.7.4.0 ❌ Exception in FO :
Screenshot 2023-01-16 at 11 33 41

Could you check for the latest version ?
I have the same issue as @khouloudbelguith .

Thanks!

File: ps_specials.php [hunk 2]
```diff
@@ -227,15 +226,27 @@ private function getSpecialProducts()
...
-        $presenter = new ProductListingPresenter(
-            new ImageRetriever(
-                $this->context->link
-            ),
-            $this->context->link,
-            new PriceFormatter(),
-            new ProductColorsRetriever(),
-            $this->context->getTranslator()
-        );
+        if (version_compare(_PS_VERSION_, '1.7.4', '>=')) {
```
Comment (Hlavtox) [inline ps_specials.php:229]:
```suggestion
        if (version_compare(_PS_VERSION_, '1.7.5', '>=')) {
```

1.7.4 is still using the old path, change the condition and it should work ;-)

**Comment by Hlavtox** [comment]
@Progi1984 Commited to the branch, hope you don't mind. So we can get this out of the way and release it. :-)
The file was moved on 1.7.5 higher - see https://github.com/PrestaShop/PrestaShop/blob/1.7.4.4/src/Core/Product/ProductListingPresenter.php

**Review by florine2623 — APPROVED** [review]
Hello @Hlavtox ,

The module works well on version before 1.7.4.0 and in recent versions !

PS 1.7.3.2 ✅
PS 1.7.4.0 ✅ 
PS 1.7.5.2 ✅
PS 1.7.8.8 ✅
PS 8.0.1 ✅
PS develop ✅


Thanks !!

**Pull request merged**